viadot/flows/genesys_to_adls.py

```diff
@@ -26,13 +31,57 @@ def add_timestamp(files_names: List = None, path: str = "", sep: str = "\t") ->
         df_updated.to_csv(os.path.join(path, file), index=False, sep=sep)
 
 
+@task(timeout=3600)
+def filter_userid(
+    files_names: list = None,
+    path: str = "",
+    sep: str = "\t",
+    userids: list = None,
+    apply_method: bool = False,
+) -> None:
+    """filter out the data frame by user ID.
+
+    Args:
+        files_names (list, optional): All file names of downloaded files. Defaults to "\t".
+        path (str, optional): Relative path to the file. Defaults to empty string.
+        userids (list, optional): List of all user IDs to select in the data frame. Defaults to None.
+        sep (str, optional): Separator in csv file. Defaults to None.
+        apply_method (bool, optional): Use this method or avoid its execution. Defaults to False.
+    """
+
+    if apply_method:
+        for file in files_names:
+            df = pd.read_csv(os.path.join(path, file), sep=sep)
+
+            # first: it gets all the conversations ID where an agent is present.
+            conversations_id = np.array([])
```

Comment:
yes because I use a numpy method named `np.unique()` very useful, otherwise I have to do it by myself in a new function when It is already done and probably more efficient.

Reply:
For getting unique values you can use `set()` - I'm not sure if this is a good solution in this case, but if so, you can use a solution like this